### PR TITLE
JIT: Streamline TreeLifeUpdater

### PR DESCRIPTION
**Pull request by jakobbotsch**
TreeLifeUpdater at most needs to flip 4 bits in two liveness sets when it is called, however before this change it uses several full width bitset operations to do this. This changes TreeLifeUpdater to do its job in a much more direct way by updating the liveness sets directly.

### CODE REVIEW

File: src/coreclr/jit/treelifeupdater.cpp [hunk 1]
```diff
@@ -188,240 +135,166 @@ void TreeLifeUpdater<ForCodeGen>::UpdateLifeVar(GenTree* tree, GenTreeLclVarComm
...
-#endif // DEBUG
-
-                if (isBorn)
+                if (isMultiRegLocal)
                 {
-                    VarSetOps::UnionD(compiler, compiler->codeGen->gcInfo.gcVarPtrSetCur, gcTrkStkDeltaSet);
+                    isFieldDying = lclVarTree->IsLastUse(i);
+                    // TODO: Remove this condition which disallows marking
+                    // some fields as dead even though they are dying when other
+                    // fields are defined.
```
Comment (jakobbotsch) [inline src/coreclr/jit/treelifeupdater.cpp:251]: I will clean up these TODOs and unify some of these cases in #80501.

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @BruceForstall

[No ASM diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=136662&view=ms.vss-build-web.run-extensions-tab) but around -1% throughput improvement on optimized collections.

The review might not be that easy as much of the logic was reworked and restructured. The old logic was using several temporary bit sets (`varDeltaSet`, `stackVarDeltaSet`, `newLife`) which necessitated full bit set operations to propagate the bits into `compCurLife` and the stack pointer set. This change just modifies the sets directly and does the other necessary updates inline too (e.g. opening/closing variable live ranges).

Will run some stress jobs too.

**Comment by jakobbotsch** [comment]
GC stress failures are unrelated and will be fixed by #80630.

**Comment by jakobbotsch** [comment]
I can reproduce the libraries-jitstress failure on main too so it also seems to be unrelated. Will try to bisect it and open an issue.

**Comment by jakobbotsch** [comment]
Ping @BruceForstall

**Review by BruceForstall — APPROVED** [review]
LGTM. Awesome TP improvement!

**Pull request merged**